### PR TITLE
[DISCO-2474] Merino: Titles in Wikipedia Suggestions Have Trailing Punctuation

### PR DESCRIPTION
**Pull request by taddes**
## References

JIRA: [DISCO-2474](https://mozilla-hub.atlassian.net/browse/DISCO-2474)
GitHub: [#366 ](https://github.com/mozilla-services/merino-py/issues/366)

## Description
Reported on the #contextual-services-tech Slack channel on June 28th 2023

Some titles on wikipedia seem to have a comma at the end (this one is likely because the title is “Split, Croatia”). It’s low priority but we should fix it at some point if we can if there are multiple occurrences.

Multiple occurrences found, mostly with geographic locations.
Screen Shot 2023-07-11 at 6 52 27 PM
Screen Shot 2023-07-11 at 7 00 27 PM

As discussed in ticket, we could just again return the title, but upon investigation this method required the removal of the comma from passed in title, as `get_best_keyword` method splits on the space, which is inclusive of the comma.  Have added logic to remove any and all punctuation that could also cause this issue. Just returning the title and then the text to the immediate right being the same (just possibly lowercase) seems like it would be cluttered and redundant. 

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2474]: https://mozilla-hub.atlassian.net/browse/DISCO-2474?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: merino/providers/wikipedia/backends/elastic.py [hunk 1]
```diff
@@ -23,7 +24,7 @@ def get_best_keyword(q: str, title: str):
     """Try to get the best autocomplete keyword match from the title. If there are no matches,
     then return the full title as a match. Lowercase everything.
     """
-    title = title.lower()
+    title = title.lower().translate(str.maketrans("", "", string.punctuation))
```
Comment (ncloudioj) [inline merino/providers/wikipedia/backends/elastic.py:27]: Translating the entire string might be surprising, e.g. `spider-man` will be translated into "spiderman" which is vastly different from the original keyword, searching the latter won't lead to the result of the former. How about we just strip the trailing punctuation of the full keyword?

```py
    return title[start_index:end_index].rstrip(string.punctuation)
```

**Review by ncloudioj — CHANGES_REQUESTED** [review]
Could you also add a test for this?

File: merino/providers/wikipedia/backends/elastic.py (hunk 1 above)
Comment (taddes) [reply merino/providers/wikipedia/backends/elastic.py]: Wfm, makes sense. Good thought @ncloudioj 🕷🦸‍♂️

Comment (ncloudioj) [reply merino/providers/wikipedia/backends/elastic.py]: Can you add a quick test for this, please?

**Review by ncloudioj — APPROVED** [review]
Looks good, having a test case for this would be nice.

Thanks!

**Comment by taddes** [comment]
Closes #366

**Pull request merged**